backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/predicate/impl/LuceneQueryStringPredicate.java

```diff
@@ -145,6 +148,74 @@ private static QueryParser.Operator toOperator(BooleanOperator operator) {
 					throw new AssertionFailure( "Unknown boolean operator: " + operator );
 			}
 		}
+	}
+
+	private static class HibernateSearchMultiFieldQueryParser extends MultiFieldQueryParser {
+		private final Map<String, LuceneCommonQueryStringPredicateBuilderFieldState> fieldStates;
+		private final LuceneSearchIndexScope<?> scope;
+
+		public HibernateSearchMultiFieldQueryParser(Analyzer analyzer, Map<String, Float> boosts,
+				Map<String, LuceneCommonQueryStringPredicateBuilderFieldState> fieldStates, LuceneSearchIndexScope<?> scope) {
+			super( boosts.keySet().toArray( String[]::new ), analyzer, boosts );
+			this.fieldStates = fieldStates;
+			this.scope = scope;
+		}
+
+		@Override
+		protected Query newFieldQuery(Analyzer analyzer, String field, String queryText, boolean quoted)
+				throws ParseException {
+			var state = fieldStates.get( field );
+
+			if ( !state.field().type().valueClass().isAssignableFrom( String.class ) ) {
+				var builder = state.field().queryElement( PredicateTypeKeys.MATCH, scope );
+				builder.value( queryText, ValueConvert.PARSE );
+
+				return LuceneSearchPredicate.from( scope, builder.build() ).toQuery( contextForField( state ) );
+			}
+
+			return super.newFieldQuery( analyzer, field, queryText, quoted );
```

Comment:
I've kept delegating to original parser logic for strings since, from what it seems, there's more to it than just a simple ` return new MatchSomethingQuery(...)`.